### PR TITLE
add mobile route for organization libraries

### PR DESCRIPTION
**Pull request by camhashemi**
@andrewconner @leogrim @ryanpbrewster @stkem 

it's much simpler to add endpoints when they're shared :) when something (required) changes, that's the hard part. interested in how you all feel about this.

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/LibraryImageController.scala [hunk 1]
```diff
@@ -33,6 +33,7 @@ class LibraryImageController @Inject() (
     extends UserActions with LibraryAccessActions with ShoeboxServiceController {
 
   def uploadLibraryImage(pubId: PublicId[Library], imageSize: Option[String] = None, posX: Option[Int] = None, posY: Option[Int] = None) = (UserAction andThen LibraryOwnerAction(pubId)).async(parse.temporaryFile) { request =>
+    // TODO(cam): this handler is used by mobile and needs to be versioned in case required fields are added or removed
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/LibraryImageController.scala:36]: This isn't needed if you have a mobile controller that calls it.

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/OrganizationController.scala [hunk 2]
```diff
@@ -90,6 +90,7 @@ class OrganizationController @Inject() (
   }
 
   def getOrganizationLibraries(pubId: PublicId[Organization], offset: Int, limit: Int) = OrganizationAction(pubId, authTokenOpt = None, OrganizationPermission.VIEW_ORGANIZATION) { request =>
+    // TODO(cam): this handler is used by mobile and needs to be versioned in case required fields are added or removed
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/OrganizationController.scala:93]: Seems like this comment saves 3 lines in a mobile controller (just copying this method). If they need a change later, preso, no extra funkiness needed

**Comment by andrewconner** [comment]
Commander logic can _usually_ be shared with no issues, that's good. Controller sharing gets a bit hairier, since there's nothing helping us keep it consistent. For such a smaller controller method, I'd just throw a copy in the mobile controller.

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/LibraryImageController.scala (hunk 1 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/LibraryImageController.scala:36]: Then we have to maintain a new controller with mostly the same methods. If we have one controller, we can create duplicate methods as we need them, but not before.

**Comment by ryanpbrewster** [comment]
I think you're on to something here, but this is not quite it.

If we had a single controller for each commander, but heavily versioned, and the version was part of the `request`, we could have the same logic but different serializations and it would all kind of work out.

**Comment by ryanpbrewster** [comment]
I'm super wary of contract-by-comment, though. The compiler doesn't tell us when we are dumb in comments.

**Comment by camhashemi** [comment]
That was sort of the idea behind `request.source` in #17041 looking at `request.path` for client info, could be extended to grab the API version.

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/LibraryImageController.scala (hunk 1 above)
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/LibraryImageController.scala:36]: The commander is typed, though, and isn't speaking `JsValue => JsValue`. You wouldn't accidentally change formats here.

**Comment by camhashemi** [comment]
But alas, I bow my head to you who have more years and wisdom than I.

**Comment by andrewconner** [comment]
If we want to unify API endpoints, we can (and should get everyone on board), but let's do it by unifying APIs, not increasing implicitness.

**Comment by camhashemi** [comment]
Got it, thanks for taking the time!

**Pull request closed without merging**